l1-contracts/test/sparta/Sparta.t.sol

```diff
@@ -64,27 +64,58 @@ contract SpartaTest is DecoderBase {
     }
   }
 
-  function _testProposerForFutureEpoch() public {
-    // @todo Implement
+  function testProposerForNonSetupEpoch(uint8 _epochsToJump) public {
+    uint256 pre = rollup.getCurrentEpoch();
+    vm.warp(block.timestamp + uint256(_epochsToJump) * rollup.EPOCH_SIZE() * rollup.SLOT_SIZE());
+    uint256 post = rollup.getCurrentEpoch();
+    assertEq(pre + _epochsToJump, post, "Invalid epoch");
+
+    address expectedProposer = rollup.getCurrentProposer();
+
+    // Add a validator which will also setup the epoch
+    rollup.addValidator(address(0xdead));
+
+    address actualProposer = rollup.getCurrentProposer();
+    assertEq(expectedProposer, actualProposer, "Invalid proposer");
   }
 
-  function _testValidatorSetLargerThanCommittee() public {
-    // @todo Implement
+  function testValidatorSetLargerThanCommittee(bool _insufficientSigs) public {
+    _testBlock("mixed_block_1", false, 0, false); // We run a block before the epoch with validators
+
+    // Adding more validators!
+    for (uint256 i = 5; i < 100; i++) {
+      uint256 privateKey = uint256(keccak256(abi.encode("validator", i)));
+      address validator = vm.addr(privateKey);
+      privateKeys[validator] = privateKey;
+      rollup.addValidator(validator);
+    }
+
+    assertGt(rollup.getValidators().length, rollup.TARGET_COMMITTEE_SIZE(), "Not enough validators");
+
+    // We should not be passing here, something should be breaking! Why is the committe small?
```

Comment:
I don't understand the comment. Are you saying something is broken currently?

Reply:
Or, probably a leftover comment.